GoogleForCreators/web-stories-wp
# Media: Ensure duration is always present on resources

## Context



## Summary

When media picker is used, it does not current return length as a field in the admin ajax call. 

In this PR, I added media_details key to the response. This contains meta data that is useful in the editor. The REST API already returns this data. So keep admin ajax / rest api, I returned the data in the exact same format. This makes the code a little simpler in a number of places and keep the data the same, in case more data is needed in the future. 

I also noticed another issue with VideoPress. Length was never being returned here, as the field was not present in the api. The tweaks the videopres integration, make the code simplier, as media details are also used here, but also makes sure that length is returned. 

There is a migration used to populate missing data and this includes a unit test. 

## Relevant Technical Choices



## To-do



## User-facing changes



## Testing Instructions



### QA






This PR can be tested by following these steps:

1. Create a new story with a few pages
2. Set page advancement duration to something like 2s
2. On the first page, insert a pre-existing video via the WP media modal (click on the "Upload" button)
3. Preview the story on the frontend
4. Verify that the page with the video on it advances only after video has finished, not after 2s

### UAT







This PR can be tested by following these steps:

1.

## Reviews

### Does this PR have a security-related impact?



### Does this PR change what data or activity we track or use?



### Does this PR have a legal-related impact?



## Checklist




---



Fixes #5030
